### PR TITLE
Allow promotion coupon to be filtered by promotion code

### PR DESCRIPTION
**Pull request by TheMilek**
| Q               | A
|-----------------|-----
| Branch?         | api-platform-3 
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| License         | MIT

### CODE REVIEW

**Review by mpysiak — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml [hunk 1]
```diff
@@ -314,6 +314,20 @@
...
+            </argument>
+            <tag name="api_platform.filter" />
+        </service>
+
+        <service id="sylius_api.search_filter.promotion_coupon_promotion" parent="api_platform.doctrine.orm.search_filter" public="true">
+            <argument type="collection">
+                <argument key="promotion.code">exact</argument>
+            </argument>
+            <tag name="api_platform.filter" />
+        </service>
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml]:
```suggestion
        <service id="sylius_api.search_filter.promotion_coupon" parent="api_platform.doctrine.orm.search_filter" public="true">
            <argument type="collection">
                <argument key="code">exact</argument>
                <argument key="promotion.code">exact</argument>
            </argument>
            <tag name="api_platform.filter" />
        </service>
```

Comment (vvasiloi) [reply src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml]: It is possible to have a promotion coupon with a code identical to a different promotion, so the filters should be separate.

Comment (Wojdylak) [reply src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml]: There isn't an ADR file for that yet, but we have decided to create one filter per resource with one or many arguments. This approach makes it easier to change in the end application.

**Review by Wojdylak — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml (hunk 1 above)
Comment (vvasiloi) [reply src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml]: Is it only for the "search" filter?

Comment (Wojdylak) [reply src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml]: It includes all API platform filters such as `search_filter`, `order_filter`, `boolean_filter`, `date_filter` itp.

**Pull request merged**